ntoskrnl/po/power.c

```diff
@@ -1118,6 +1119,16 @@ NtSetSystemPowerState(IN POWER_ACTION SystemAction,
         break;
     }
 
+    /*
+     * TODO: Investigate in the future why Windows Server 2003
+     * needs to get the NT product type during a system power state
+     * action.
+    */
+    if (!RtlGetNtProductType(&ProductType))
```

Comment:
As long as we do not know why MS does this call, I see absolutely no benefit in adding it here.
Stuff that is called without any reason is wasted time, bloated binary size, and additional risk just, especially in the kernel.
If you absolutely feel the need to add it to rememeber that MS does call it here, then please at least comment **all the code** that you added within power.c!

Reply:
Agreed, we don't need dead code.
FYI, the reason Windows does this seems to have something to do with writing event log entries during hibernation.

Reply:
Yeah, apologies for that. I'll address the comment reviews in the next force push.